include/rocksdb/utilities/object_registry.h

```diff
@@ -38,6 +39,30 @@ using FactoryFunc =
 // library
 using RegistrarFunc = std::function<int(ObjectLibrary&, const std::string&)>;
 
+// The signature of the function for loading plugins
+// The Plugin struct is passed to the PluginFunc and expected to be validated
+// and filled in by the Plugin code.
+// The Plugin contains the struct of properties associated with this Plugin
```

Comment:
Plugin doesn't have such properties, nor does it appear to be meaningfully extensible.

Reply:
> 

I am not sure what you are referring to, as your comment does not line up with the code. 

The RegistrarFunc takes in an ObjectLibrary and an "arg".  This arg was meant to be able to change how classes were registered.  For example,  the RegistrarFunc could register objects in "test" mode or "debug" mode based on this argument.  In practice, most cases do not use this argument at all (though I could imagine wanting to use it to prevent some test classes -- like a ROT13 cipher perhaps -- from ending up in production.